accounts-db/src/accounts_db.rs

```diff
@@ -2862,6 +2862,19 @@ impl AccountsDb {
                                         } else {
                                             found_not_zero += 1;
                                         }
+
+                                        // If this candidate has multiple rooted slot list entries,
+                                        // we should reclaim the older ones.
+                                        if slot_list.len() > 1
+                                            && *slot
```

Comment:
What about unrooted slots in a fork?

Reply:
Never mind. I don't think we ever pass those fork slots to clean. It should happen.

Reply:
Imaging that we are passing `max_clean_root_inclusive=None` to this clean function. 

In the old code, we are protected by `self.uncleaned_roots()` even when we pass `max_clean_root_inclusive=None`, we wont' lose any roots because of unrooted forks. 

However, now with this change, it will be wrong if we pass `max_clean_root_inclusive=None`  and there are unrooted forks. Clean will drop the last rooted (pubkey, slots) because there is a newer unrooted entry. 

I think with this change, we must require to pass `max_clean_root_inclusive`, otherwise the call may have incorrect behavior. 

For validator, which have snapshot disabled, we may pass `max_clean_root_inclusive=None` to clean and we will lost any roots because of unrooted forks.

Reply:
Ah. Yes, your are right. We are good. 

Thanks for adding the tests.